internal/subscription/service/course_service.go

```diff
@@ -0,0 +1,40 @@
+package service
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/sumelms/microservice-course/internal/course"
+	courseDomain "github.com/sumelms/microservice-course/internal/course/domain"
+)
+
+/*
+	I create this service which will be responsible to communicate with
+	the course package. In the future, if we move the subscription or course to another service,
+	this will be the only point that we will need to change.
+*/
+
+type courseSvc struct {
+	course courseDomain.ServiceInterface
+}
+
+func NewCourseSvc(db *sqlx.DB, logger log.Logger) courseSvc {
+	/*
+		This is not the best way to do that, because we are creating a new course service
+		which is already created when we run the application instead of receiving a
+		already created service by param.
+		But with this architecture it was the best way to do it.
+	*/
+	svc := course.NewService(db, logger)
+	return courseSvc{
+		course: svc,
+	}
+}
+
+func (svc courseSvc) ExistCourse(ctx context.Context, id uuid.UUID) error {
```

Comment:
@canhizares I'm curious here. In this case we don't actually return any model from the database, right? So, what would happend if we have something more like a get action, that retrieves the Course object into the subscription domain? Where the Course model will live? It will stay at the `course` pkg or the `subscription` pkg must have its own model definition for it?

Reply:
@rluders In that case, the Course model should be in the `subscription` package, because otherwise, we would have coupling again.
And, the `subscription.Course` does not have to be identical to `course.Course`, may be subscription needs lees information. 

an example could be:

`func (svc courseSvc) GetCourse(ctx context.Context, id uuid.UUID) (domain.Course, error) {}`

And this GetCourse will be responsible (and the only one that will know how is the `course.Course`)  to receive the `course.Course` and map it to the `subscription.Course`

Reply:
Perfect. That was exactly what I was thinking.